collector/mountstats_linux.go

```diff
@@ -538,20 +538,29 @@ func (c *mountStatsCollector) Update(ch chan<- prometheus.Metric) error {
 			mountAddress = miStats.SuperOptions["addr"]
 		}
 
-		deviceIdentifier := nfsDeviceIdentifier{m.Device, stats.Transport.Protocol, mountAddress}
+		deviceIdentifier := nfsDeviceIdentifier{m.Device, lastTransportOrEmpty(stats.Transport).Protocol, mountAddress}
 		i := deviceList[deviceIdentifier]
 		if i {
 			level.Debug(c.logger).Log("msg", "Skipping duplicate device entry", "device", deviceIdentifier)
 			continue
 		}
 
 		deviceList[deviceIdentifier] = true
-		c.updateNFSStats(ch, stats, m.Device, stats.Transport.Protocol, mountAddress)
+		c.updateNFSStats(ch, stats, m.Device, lastTransportOrEmpty(stats.Transport).Protocol, mountAddress)
 	}
 
 	return nil
 }
 
+// lastTransportOrEmpty returns last element from given slice. Before https://github.com/prometheus/procfs/issues/450,
```

Comment:
Hmm, I wonder if we should handle this better, now that we have all the data.

Reply:
Better would be to export each stat individually, but we would need some new label for that -- perhaps "port"?

Reply:
PTAL at https://github.com/prometheus/node_exporter/pull/3045/commits/5237827256b2a57a5dde6665c6d27c2fc30b098b, where I've added `port` label to transport-metrics.

Reply:
I think we should also remove `protocol` from "global" metrics, since protocol is per transport. But that can break some queries.